core/src/main/java/apoc/refactor/GraphRefactoring.java

```diff
@@ -452,6 +458,76 @@ public void categorize(
         }
     }
 
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.refactor.deleteAndReconnect([pathLinkedList], [nodesToRemove], {config}) - Removes some nodes from a linked list")
+    public Stream<GraphResult> deleteAndReconnect(@Name("path") Path path, @Name("nodes") List<Node> nodesToRemove, @Name(value = "config", defaultValue = "{}") Map<String,Object> config) {
+
+        RefactorConfig refactorConfig = new RefactorConfig(config);
+
+        List<Node> nodes = IterableUtils.toList(path.nodes());
+        Set<Relationship> rels = Iterables.asSet(path.relationships());
+
+        if (!nodes.containsAll(nodesToRemove)) {
+            throw new RuntimeException("Some nodes in nodesToRemove not present in the path");
+        }
+
+        BiFunction<Node, Direction, Relationship> filterRel = (node, direction) -> StreamSupport
+                .stream(node.getRelationships(direction).spliterator(), false)
+                .filter(rels::contains)
+                .findFirst()
```

Comment:
to be consistent with the first strict check, shouldn't an error be raised if more for than 1 relationship matches?

Reply:
Now I have removed the first error. Therefore the consistence (that is, raise a `RuntimeException` or other) would no longer exist. I dunno, maybe it's better not to throw the exception here?